src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/Expressions/ExpressionResolverBuilder.cs

```diff
@@ -100,7 +100,9 @@ public ExpressionResolverBuilder(CallSiteRuntimeResolver runtimeResolver, IServi
                     ScopeParameter);
             }
 
-            return Expression.Lambda<Func<ServiceProviderEngineScope, object>>(VisitCallSite(callSite, null), ScopeParameter);
+            return Expression.Lambda<Func<ServiceProviderEngineScope, object>>(
+                Convert(VisitCallSite(callSite, null), typeof(object), forceValueTypeConversion: true),
```

Comment:
This isn't required.

Reply:
My new tests fail without it.

Reply:
I see, that makes me think this repros in 3.x too?

Reply:
I mean before 3x

Reply:
The original bug was for IL Emit. The way the tests are set up, they test the scenario for all the engine types.

So yes, for LinqExpressions, this probably reproed in 3.x and before 3.x as well. I assume no one ever tried it before.